src/main/java/com/khaliuk/model/Product.java

```diff
@@ -1,16 +1,27 @@
 package com.khaliuk.model;
 
 public class Product {
+    private static Long idCounter = 0L;
```

Comment:
idCouter should not be stored here, it is not responsibility of Product to store this information